### ISSUE TITLE
CI: Invalid stdout/stderr output printed after executing commands

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#5295]
I think the following commit introduced a regression in the CI:
```
commit 16ef796970ccf7b2d1af7fe2fe436bb734fe05d5
Author: Eloy Coto <eloy.coto@gmail.com>
Date:   Fri Aug 17 17:29:09 2018 +0200

    Test: Add curlFail with statistics

    Enable statistics on CurlFail helpers so it'll be easy to debug if is a
    DNS issue or a connectivity issue.

    Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
```

master build failures:
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1457/
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1458/

Related issue:
#5292

**Comment by eloycoto** [comment]
Hi,

Failed test for the track:

1) https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1458/testReport/(root)/runtime/RuntimePolicies_TestsEgressToHost_Tests_egress_with_CIDR_L7_policy/
2) https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1457/testReport/junit/(root)/runtime/RuntimePolicies_TestsEgressToHost_Tests_egress_with_CIDR_L7_policy/
3) https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/3752/testReport/junit/(root)/runtime/RuntimePolicies_TestsEgressToHost_Tests_egress_with_CIDR_L7_policy/

Both output has a corrupted chars:

```
/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
unexpectedly able to access http://"172.17.0.1":80/private when access should only be allowed to /index.html
Expected
    <string>: ��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
to contain substring
    <string>: 403
/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/runtime/Policies.go:1304

```

```
/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
unexpectedly able to access http://"172.17.0.1":80/private when access should only be allowed to /index.html
Expected
    <string>: ��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
to contain substring
    <string>: 403
/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/runtime/Policies.go:1304
```


```
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
unexpectedly able to access http://"172.17.0.1":80/private when access should only be allowed to /index.html
Expected
    <string>: ��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
to contain substring
    <string>: 403
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:1304
```

I can discard that there is any other test that keep Cilium in a bad state, in
this build RuntimePolicies are the first test:

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/3752/execution/node/96/log/?consoleFull

Need to say that we also saw the invalid characters on monitor output.

All related to egress traffic looks legit from Envoy:

```
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="Queueing container event" containerID=d9f468d7cb event="exec_create: curl -s --fail --output /dev/stderr -w %{http_code} --connect-timeout 3 http://172.17.0.1/private" subsys=workload-watcher
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="Queueing container event" containerID=d9f468d7cb event="exec_start: curl -s --fail --output /dev/stderr -w %{http_code} --connect-timeout 3 http://172.17.0.1/private" subsys=workload-watcher
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[cilium_socket_option.h:55] Cilium SocketOption(): source_identity: 38008, destination_identity: 54404, ingress: false, port: 80, proxy_port: 13793" subsys=envoy-filter threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[cilium_network_filter.cc:55] Cilium Network: onNewConnection" subsys=envoy-filter threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[cilium_network_filter.cc:69] [C114] Cilium Network: Added connection callbacks to delete proxymap entry later" subsys=envoy-filter threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/server/connection_handler_impl.cc:217] [C114] new connection" subsys=envoy-main threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/common/http/conn_manager_impl.cc:191] [C114] new stream" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/common/http/conn_manager_impl.cc:848] [C114][S5168545385248448759] request end stream" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/common/http/conn_manager_impl.cc:488] [C114][S5168545385248448759] request headers complete (end_stream=true):" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="':authority', '172.17.0.1'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="':path', '/private'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="':method', 'GET'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="'user-agent', 'curl/7.52.1'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="'accept', '*/*'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[cilium_l7policy.cc:154] Cilium L7: Egress (38008->54404) policy lookup for endpoint 10.15.27.114: DENY" subsys=envoy-filter threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/common/http/conn_manager_impl.cc:1042] [C114][S5168545385248448759] encoding headers via codec (end_stream=false):" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="':status', '403'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="'content-length', '15'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="'content-type', 'text/plain'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="'date', 'Tue, 21 Aug 2018 19:14:29 GMT'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="'server', 'envoy'" subsys=envoy-http threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="10.15.27.114: Access log message: timestamp:1534878869649726432 http_protocol:HTTP11 entry_type:Denied policy_name:\"10.15.27.114\" source_security_id:38008 source_address:\"10.15.27.114:57368\" destination_address:\"172.17.0.1:80\" scheme:\"http\" host:\"172.17.0.1\" path:\"/private\" method:\"GET\" status:403 headers:<key:\"user-agent\" value:\"curl/7.52.1\" > headers:<key:\"accept\" value:\"*/*\" > headers:<key:\"x-request-id\" value:\"2ba93ee5-8e25-4bb2-9c89-6a7cc09934d2\" > " subsys=envoy-manager
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="Logging flow record" code=403 header="map[Accept:[*/*] X-Request-Id:[2ba93ee5-8e25-4bb2-9c89-6a7cc09934d2] User-Agent:[curl/7.52.1]]" message= method=GET protocol=HTTP/1.1 subsys=proxy-logger type=Request url="http://172.17.0.1/private" verdict=Denied
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="Skipping writing to access log (logger nil)" file-path= subsys=proxy-logger
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/common/network/connection_impl.cc:451] [C114] remote close" subsys=envoy-connection threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/common/network/connection_impl.cc:133] [C114] closing socket: 0" subsys=envoy-connection threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[cilium_network_filter.cc:94] [C114] Cilium Network: Connection Closed, proxymap cleanup succeeded" subsys=envoy-filter threadID=18837
Aug 21 19:14:29 runtime cilium-agent[16268]: level=debug msg="[external/envoy/source/server/connection_handler_impl.cc:50] [C114] adding to cleanup list" subsys=envoy-main threadID=18837
```

**Comment by eloycoto** [comment]
I tested again, and this issue is reproducible with the following commands: 
```
diff --git a/ginkgo.Jenkinsfile b/ginkgo.Jenkinsfile
index 0d4e55573..45e3f5874 100644
--- a/ginkgo.Jenkinsfile
+++ b/ginkgo.Jenkinsfile
@@ -97,7 +97,7 @@ pipeline {
                 script {
                     parallel(
                         "Runtime":{
-                            sh 'cd ${TESTDIR}; ginkgo --focus=" Runtime*" --seed=1534775667 -v --failFast=${FAILFAST}'
+                            sh 'cd ${TESTDIR}; ginkgo --focus=" Runtime*" -v --failFast=${FAILFAST}'
                         },
                         "K8s-1.8":{
                             sh 'cd ${TESTDIR}; K8S_VERSION=1.8 ginkgo --focus=" K8s*" -v --failFast=${FAILFAST}'
diff --git a/test/helpers/wrappers.go b/test/helpers/wrappers.go
index 173594656..24f6e2984 100644
--- a/test/helpers/wrappers.go
+++ b/test/helpers/wrappers.go
@@ -53,7 +53,7 @@ func CurlFail(endpoint string, optionalValues ...interface{}) string {
 		endpoint = fmt.Sprintf(endpoint, optionalValues...)
 	}
 	return fmt.Sprintf(
-		`curl -s -D /dev/stderr  --trace /dev/stderr --fail --connect-timeout %[1]d --max-time %[1]d %[2]s -w "%[3]s"`,
+		`curl -s -D /dev/stderr --fail --connect-timeout %[1]d --max-time %[1]d %[2]s -w "%[3]s"`,
 		CurlConnectTimeout, endpoint, statsInfo)
 }
 
@@ -67,7 +67,7 @@ func CurlWithHTTPCode(endpoint string, optionalValues ...interface{}) string {
 	}
 
 	return fmt.Sprintf(
-		`curl -s --fail --output /dev/stderr --trace /dev/stderr -w '%%{http_code}' --connect-timeout %d %s`,
+		`curl -s --fail --output /dev/stderr -w '%%{http_code}' --connect-timeout %d %s`,
 		CurlConnectTimeout, endpoint)
 }
 
```
Evidence: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5808/execution/node/91/log/?consoleFull
PR: https://github.com/cilium/cilium/pull/5302

**Comment by eloycoto** [comment]
Fixed by #5319
